src/pip/_internal/cache.py

```diff
@@ -157,7 +158,9 @@ def get_path_for_link(self, link):
         parts = self._get_cache_path_parts(link)
 
         # Store wheels within the root cache_dir
-        return os.path.join(self.cache_dir, "wheels", *parts)
+        return os.path.join(
+            self.cache_dir, implementation_tag, "wheels", *parts
```

Comment:
Instead of putting the implementation tag here can we use it in the cache key hash computation?

Also it would be good to use generic system values, like `'.'.join(sys.version_info[:2])` and `sys.implementation.name` instead of `pep425tags.implementation_tag`, since the particular format doesn't matter and the latter is going away.

Reply:
> can we use it in the cache key hash computation?

@chrahunt Probably yes. I thought it could be useful to have a human-readable subtree in case someone would need to do some manual cache pruning (although I'd change it to read `{cache_dir}/wheels/{impl}`). Are there other advantages/disadvantages to one or the other approach that you would have in mind?

Reply:
It's simpler IMO, since any change to the criteria we use will always just result in an update to the cache key. Compare to "most things cause an update to the cache key except implementation which changes the directory". It also avoids having to reason about the order in which criteria are applied or worry whether we're getting close to `MAX_PATH` on Windows.

Later if we want introspection ability we could add a json file to the directory with the cache key details.